### PR TITLE
feature_release logging:108 implement logging and remove bugs that occur by logging

### PR DESCRIPTION
**Pull request by daniwe4**
https://github.com/conceptsandtraining/doil/issues/108

### CODE REVIEW

File: app/src/Lib/SymfonyShell.php [hunk 1]
```diff
@@ -4,39 +4,38 @@
...
 
-    protected function runTTYQuiet(array $commands) : void
-    {
-        $process = new Process($commands);
-        $process->setTty(true);
-        $process->setTimeout(36000);
-        $process->disableOutput();
-        $process->run();
+        // executes after the command finishes
+        if (! $process->isSuccessful()) {
```
Comment (tbongers-cat) [inline app/src/Lib/SymfonyShell.php:36]: Please remove blank space after '!'

**Review by tbongers-cat — COMMENTED** [review]
Some small adjustments

File: app/src/Lib/Docker/DockerShell.php [hunk 2]
```diff
@@ -343,7 +397,9 @@ public function stop(string $name) : void
             $name
         ];
 
-        $this->run($cmd);
+        $logger = $this->logger->getDoilLogger($name);
+        $this->run($cmd, $logger);
+        $logger->info("Stop instance");
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:402]: The name is already the parameter for getDoilLogger.

File: app/src/Lib/Docker/DockerShell.php [hunk 3]
```diff
@@ -37,20 +45,22 @@ public function startContainerByDockerCompose(string $path) : void
             "--force-recreate"
         ];
 
-        $this->run($cmd);
+        $logger = $this->logger->getDoilLogger(pathinfo($path, PATHINFO_FILENAME));
+        $this->run($cmd, $logger);
+        $logger->info("Start instance");
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:50]: Thats right, done!

Instance name is given by getDoilLogger

File: app/src/Lib/Docker/DockerShell.php [hunk 4]
```diff
@@ -384,7 +444,9 @@ public function setGrain(string $name, string $key, string $value) : void
             "salt \"$name\" grains.setval \"$key\" \"$value\""
         ];
 
-        $this->runTTY($cmd);
+        $logger = $this->logger->getSaltLogger($name);
+        $this->runTTY($cmd, $logger);
+        $logger->info("Set grain key '$key' with value '***' for insatnce '$name'");
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:449]: Done!

File: app/src/Lib/Docker/DockerShell.php [hunk 5]
```diff
@@ -406,9 +468,12 @@ public function pruneNetworks() : void
...
             "network",
             "prune",
-            "-f"
+            "-f",
+            "&>/dev/null"
         ];
-        $this->runTTYQuiet($cmd);
+
+        $logger = $this->logger->getDoilLogger("docker");
+        $this->runTTY($cmd, $logger);
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:476]: Done!

File: app/src/Lib/Docker/DockerShell.php [hunk 6]
```diff
@@ -458,25 +526,27 @@ protected function startContainerByDockerComposeWithForceRecreate(string $path)
             "--force-recreate"
         ];
 
-        $this->run($cmd);
+        $logger = $this->logger->getDoilLogger(pathinfo($path, PATHINFO_FILENAME));
+        $this->run($cmd, $logger);
+        $logger->info("Start instance");
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:531]: Is given by getDoilLogger

File: app/src/Lib/Logger/LoggerFactory.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+namespace CaT\Doil\Lib\Logger;
+
+use Monolog\Logger;
+use Psr\Log\LoggerInterface;
+use Monolog\Handler\StreamHandler;
+use Monolog\Formatter\LineFormatter;
+
+class LoggerFactory
+{
+    protected const DOIL_LOG_PATH = "/var/log/doil/doil.log";
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Logger/LoggerFactory.php:14]: Done!

File: app/src/Lib/SymfonyShell.php (hunk 1 above)
Comment (daniwe4) [reply app/src/Lib/SymfonyShell.php:36]: No. Thats part of our coding style.

Comment (daniwe4) [reply app/src/Lib/SymfonyShell.php:36]: Thats so right. Done!

**Review by tbongers-cat — APPROVED** [review]
All good now :)

**Pull request merged**